### PR TITLE
Complete testing and documentation of Python 3.9 support

### PR DESCRIPTION
**Pull request by jdufresne**
Document and add support for Python 3.9.

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

---

Testing was added in 7503ad75d849b96c6e29afeefbc2d735681854ae but is now
included in all CI matrices and is documented as trove classifier.

Python 3.9 was released November 15, 2020.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,18 +21,15 @@ jobs:
...
           - 3.5
           - 3.6
           - 3.7
+          - 3.8
         pip-version:
           - "latest"
           - "20.1"
-        include:
-          - os: Ubuntu
-            python-version: 3.9-dev
```
Comment (atugushev) [inline .github/workflows/ci.yml:34]: Maybe test `3.10-dev` now? Deadsnakes supports it.

Comment (jdufresne) [reply .github/workflows/ci.yml:34]: Done!

**Review by atugushev — APPROVED** [review]
Thanks 👍 I think it would be good to remove py35 (due to EOL) in a following up PR.

**Comment by jdufresne** [comment]
> I think it would be good to remove py35 (due to EOL) in a following up PR.

Would be happy to.

What is the pip-tools policy on supported Pythons? 

If we're using EOL as a guideline, then we should begin work towards dropping Python 2. Something I'm also happy to contribute to.  See https://devguide.python.org/#status-of-python-branches

Or, on the other hand, are we trying to match the pip support? If so, they still support Python 3.5.

Or, is the policy drop EOL Pythons except 2.7?

I looked through the docs and wasn't able to find anything explicit. I could add something to <https://github.com/jazzband/pip-tools#versions-and-compatibility> once I understand the policy.

My preference is to follow Python EOL branches, but I recognize that may too fast for some projects.

**Pull request merged**